deploy/wmagent-venv-deploy.sh

```diff
@@ -0,0 +1,672 @@
+    echo "..."
+
+    wmSrcPath=${venvPath}/srv/WMCore           # WMCore source code target path
+
+    # NOTE: If the Virtual Environment is not to be cleaned during the current
+    #       deployment and we already have either a source directory synced from
+    #       previous deployments or a link at $wmSrcPath pointing to a source
+    #       directory outside the virtual env. we simply skip git actions to protect
+    #       developer's previous work.
+    if $noVenvCleanup && ( [[ -d $wmSrcPath ]] || [[ -h $wmSrcPath ]] ); then
+        echo "WMCore source has already been cloned and the NO Virtual Environment Cleanup is True."
+        return 101
+    else
+        [[ -d $wmSrcPath ]] ||  mkdir -p $wmSrcPath || return $?
+        cd $wmSrcPath
+        git clone $wmSrcRepo $wmSrcPath && git checkout $wmSrcBranch && [[ -n $wmTag ]] && git reset --hard $wmTag
+        cd -
+    fi
+}
+
+_pipUpgradeVenv(){
+    # Helper function used only for pip Upgrade for the current virtual env
+    # :param: None
+    cd $venvPath
+    # pip install $pipOpt wheel
+    # pip install $pipOpt --upgrade pip
+    pip install wheel
+    pip install --upgrade pip
+}
+
+activateVenv(){
+    # Function for activating the virtual environment
+    # :param: None
+    echo
+    echo "======================================================="
+    echo "Activate WMCore virtual env:"
+    echo -n "Continue? [y]: "
+    $assumeYes || read x && [[ $x =~ (n|no|nO|N|No|NO) ]] && return 102
+    echo "..."
+    source ${venvPath}/bin/activate
+    _pipUpgradeVenv
+}
+
+setupRucio(){
+    # Function to create a minimal setup for the Rucio package inside the virtual
+    # environment. It uses rucio integration as default server to avoid interference
+    # with production installations.
+    # :param: None
+
+    # NOTE: This configuration files will be used mostly during operations and
+    #       development or running from source. The different services usually rely
+    #       on their own configuration service_config script for defining the Rucio
+    #       instance, but we still my consider adding an extra parameter to the script
+
+    echo
+    echo "======================================================="
+    echo "Create minimal Rucio client setup inside the virtual env:"
+    echo -n "Continue? [y]: "
+    $assumeYes || read x && [[ $x =~ (n|no|nO|N|No|NO) ]] && return 101
+    echo "..."
+
+    pip install $pipOpt rucio-clients
+    _addWMCoreVenvVar RUCIO_HOME $venvPath
+
+    cat <<EOF> $RUCIO_HOME/etc/rucio.cfg
+[common]
+[client]
+rucio_host = http://cms-rucio-int.cern.ch
+auth_host = https://cms-rucio-auth-int.cern.ch
+auth_type = x509
+ca_cert = /etc/grid-security/certificates/
+client_cert = \$X509_USER_CERT
+client_key = \$X509_USER_KEY
+client_x509_proxy = \$X509_USER_PROXY
+request_retries = 3
+EOF
+}
+
+setupIpython(){
+    # Helper function to install Ipython during manual installation, it is skipped by default.
```

Comment:
It is actually installed by default, not skipped.

Reply:
comment fixed